### PR TITLE
add sha256 test case

### PR DESCRIPTION
**Pull request by mehlis**

### CODE REVIEW

File: test_sha256/main.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    SHA256_Final(hash, &sha256);
+
+    for (int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
+        printf("%02x", hash[i]);
+    }
+
+    printf("\n");
+}
+
+int main()
```
Comment (OlegHahm) [inline test_sha256/main.c]: warning: function declaration isn't a prototype

Parameter `void` is missing

**Comment by OlegHahm** [comment]
A simple test to check the ouput against python or `sha256` from GNU coreutils would be nice.

**Comment by mehlis** [comment]
I've added more test cases to this PR

**Comment by OlegHahm** [comment]
Ack, merge with https://github.com/RIOT-OS/RIOT/pull/96

**Pull request merged**